### ISSUE TITLE
Alacritty crashes on Windows 7 on startup

### ISSUE THREAD

**Issue by pvonmoradi** [jwilm/alacritty#1748]
I've downloaded the two binaries from [https://ci.appveyor.com/project/jwilm/alacritty/build/job/4ampw5l309pb0aox/artifacts](https://ci.appveyor.com/project/jwilm/alacritty/build/job/4ampw5l309pb0aox/artifacts). Not sure how to run the application but it crashed on double click. Here is the output when I use `-vvv` switch from cmd:  
```
C:\Users\Pooya\Downloads>alacritty.exe -vvv

C:\Users\Pooya\Downloads>←[33mConfig `custom_cursor_colors` is deprecated.←[0m
←[33mConfig `cursor_style` is deprecated. Please use `cursor.style` instead.←[0m
←[33mConfig `hide_cursor_when_typing` is deprecated. Please use `mouse.hide_when_typing` instead.←[0
m
←[33mConfig `unfocused_hollow_cursor` is deprecated. Please use `cursor.unfocused_hollow` instead.←[
0m
Welcome to Alacritty.
Configuration loaded from C:\Users\Pooya\alacritty.yml
device_pixel_ratio: 1
width: 1024, height: 750
Initializing glyph cache
```
(It's not exiting actually)
And some of the outputs of the crash window:
```
Problem signature:
  Problem Event Name:	APPCRASH
  Application Name:	alacritty.exe
  Application Version:	0.0.0.0
  Application Timestamp:	5be0306a
  Fault Module Name:	atio6axx.dll
  Fault Module Version:	6.14.10.9902
  Fault Module Timestamp:	4c05c596
  Exception Code:	c0000094
  Exception Offset:	0000000000928f0b

```
> Which operating system does the issue occur on?

Windows 7 x64

**Comment by chrisduerr** [comment]
@pvonmoradi Have you tried executing the winpty executable before starting Alacritty? Also are both stored in the same directory?

**Comment by pvonmoradi** [comment]
@jwilm I just checked `Alacritty.exe` and `Alacritty-binaries.exe` from releases and substitured my autogenerated .yaml in home with alacritty_windows.yml. Same thing happens.
@chrisduerr How can I do it? I run it but it shows the console and closes. Yes there are in the same directory.

**Comment by chrisduerr** [comment]
You said that it doesn't actually crash, does the cmd.exe work properly? Do you have bash on your system, maybe that will work instead.

I've been running Alacritty on Windows 7 64bit myself, so it certainly isn't an issue with the operating system in general.

**Comment by pvonmoradi** [comment]
@chrisduerr When I run it from cmd.exe or double click it, it shows the window and immoderately crash report window appears. When I close the crash report window, I see alacrity in cmd.exe has not exited.
I have cmder and MSYS2. I tried it using both bash from cmder (which is git-for-windows's bash I guess) and MSYS2's bash. 
This way the window doesn't even appear at it exits. With -vvv switch, the same text is outputed:
```
$ ./Alacritty -vvv
Welcome to Alacritty.
Configuration loaded from C:\Users\Pooya\alacritty.yml
device_pixel_ratio: 1
width: 1024, height: 750
Initializing glyph cache
```

**Comment by chrisduerr** [comment]
Is there ever any crash message in the terminal where you start Alacritty?

What messages do you get when you run `RUST_LOG=trace ./Alacritty -vvv`?

**Comment by pvonmoradi** [comment]
@chrisduerr No, only the Window's[ Alacritty's stopped working](https://0bin.net/paste/HE-RkbPX5niPzfs2#-Dn67On01ayb7mtn9PpXleO/MeHgjAGgvm0l6TUbFYV) appears when I run it from cmd or double-click it. 
I can't do that in cmd.exe but inputting that command in msys2's bash produces this:
$ RUST_LOG=trace ./Alacritty -vvv
```
 INFO 2018-11-05T20:04:14Z: alacritty: Welcome to Alacritty.
 INFO 2018-11-05T20:04:14Z: alacritty: Configuration loaded from C:\Users\Pooya\alacritty.yml
 INFO 2018-11-05T20:04:14Z: alacritty::display: device_pixel_ratio: 1
 INFO 2018-11-05T20:04:14Z: alacritty::renderer: width: 1024, height: 750
 INFO 2018-11-05T20:04:14Z: alacritty::display: Initializing glyph cache
```
No I didn't change that yaml file in my home. I even deleted it and rerun the app to make it again.

**Comment by chrisduerr** [comment]
Hmm, maybe @zacps has some ideas. It's hard to troubleshoot this without any errors. I'd assume it's probably because of some Windows subsystem like the winpty executable.

If you have mysys2's bash installed, I'd recommend trying to set Alacritty's shell to use it either with the command line parameter `-e` or the config file option.

**Comment by zacps** [comment]
Because of the lack of error message I would assume this is an exception thrown from winpty, which we can't catch or display. Could you try running it under the msvc debugger with 'break on exceptions' turned on?

**Comment by pvonmoradi** [comment]
@zacps I don't have msvc toolchain on my current machine. However I can test it on another machine or download the toolchain on this one later. Will post here when I do it. Thanks :)

**Comment by pvonmoradi** [comment]
I also tested it on Windows 10 on another machine. [Here is the error dialog](https://0bin.net/paste/pLDujq7Gp+K1aKuD#+CxnkDoNJZKX-UBUOAv517SJ/Dh4UyS50JsBy4ROU4a).
Here is the error message running it from MSYS2's bash:
$ ./Alacritty.exe -vvv

> C:/Alacritty/Alacritty.exe: error while loading shared libraries: api-ms-win-crt-heap-l1-1-0.dll: cannot open shared object file: No such file or directory

**Comment by pvonmoradi** [comment]
Installing "**Microsoft Visual C++ Redistributable for Visual Studio 2017**" for Windows 10 x64 from [this link](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads) solved the problem for windows 10.
Starting from file explorer and msys2 launches Alacritty:
`$ ./Alacritty.exe -vvv
Welcome to Alacritty.
Configuration loaded from C:\msys64\home\rozhansystem\alacritty.yml
device_pixel_ratio: 1
width: 1024, height: 749
Initializing glyph cache
Finished initializing glyph cache in 0.084400404
width: 644, height: 412
Cell Size: (8 x 17)
PTY Dimensions: Line(24) x Column(80)
Scaling RGB by 0.66 from Rgb { r: 234, g: 234, b: 234 } to Rgb { r: 154, g: 154, b: 154 }
Using config-provided dim colors
Scaling RGB by 0.66 from Rgb { r: 234, g: 234, b: 234 } to Rgb { r: 154, g: 154, b: 154 }
Using config-provided dim colors
Initialisation complete
Font size changed: Size(22)
Term::resize
Term::resize dimensions unchanged
width: 644, height: 412
Set Attribute: Reset
clear_line: Right
clear_line: Right
unset_mode: ShowCursor
carriage_return
linefeed
clear_line: Right
carriage_return
linefeed
clear_line: Right
carriage_return
linefeed
clear_line: Right
set_mode: ShowCursor
clear_line: Right
clear_line: Right
clear_line: Right
clear_line: Right
unset_mode: ShowCursor
carriage_return
linefeed
clear_line: Right
set_mode: ShowCursor
wait_object_0
wait_object_0
Goodbye.
`
**1-** I have not tested windows 7 yet. **2-** There seems to be colors and rendering problems (like in vim, midnight commander) but that is another issue.

**Comment by pvonmoradi** [comment]
I installed "Microsoft Visual C++ Redistributable for Visual Studio 2017" on my Windows 7 machine but the problem persists.
@zacps I have MSVC command line build tools but it seems its debugger is actually part of "Windows SDK package". However MS download utility did not seem to offer this tool for windows 7.
After quite a search, it turns out SDK for Windows 10 actually means SDK for windows 7, 8, 8.1 and 10. And it is 3GB! so at this point I quit trying :D

**Comment by chrisduerr** [comment]
According to @nico-abram this has been resolved. Please let me know if that's incorrect.
